Fix error when locating Lime's bundled NDLLs.

player-03 opened the pull request:
We _could_ fall back to `haxelib path` for very old versions of Haxelib, but most likely no one uses a version that old. Even if you're stuck on Haxe 3, you can still update Haxelib. Also, `haxelib path` gave incorrect results in the first place, so falling back wouldn't do any good.

I believe this will resolve #1786.

player-03 committed:
Use `haxelib libpath` when searching for NDLLs.

We could fall back to `haxelib path` for very old versions of Haxelib, but most likely even people stuck on Haxe 3 have updated Haxelib.

Also, `haxelib path` gave incorrect results in the first place, so falling back wouldn't do any good.

player-03 committed:
Fix uninitialized variable.

tobil4sk commented:
> `haxelib path` gave incorrect results in the first place

If you run `haxelib path lime` the output looks something like this:

```hxml
-L /path/to/lime/git/ndll/
--macro lime._internal.macros.DefineMacro.run()
/path/to/lime/git/src/
-D lime=8.1.1
```

The `-L` flags have the ndll path, that's the correct thing to use if it's coming from `haxelib path`.

player-03 commented:
We could do that too, as long as we rename the function to indicate that's what it's for.

tobil4sk commented:
Looks like this happens here:
https://github.com/openfl/lime/blob/d2ea41570bc195bd3c1ded9e67a2881b3cf31df2/tools/CommandLineTools.hx#L456-L462

player-03 committed:
Replace `__findHaxelib()` with a function that finds Lime's NDLL folder.

This is its only use case, so there's no need for an intermediate step of getting the library root. `haxelib path` does return the NDLL folder; the problem with the old implementation of `__findHaxelib()` was it got Lime's source folder instead.
